src/pubkey.cpp

```diff
@@ -14,158 +14,6 @@ namespace
 secp256k1_context* secp256k1_context_verify = NULL;
 }
 
-/** This function is taken from the libsecp256k1 distribution and implements
- *  DER parsing for ECDSA signatures, while supporting an arbitrary subset of
- *  format violations.
- *
- *  Supported violations include negative integers, excessive padding, garbage
- *  at the end, and overly long length descriptors. This is safe to use in
- *  Bitcoin because since the activation of BIP66, signatures are verified to be
- *  strict DER before being passed to this module, and we know it supports all
```

Comment:
In ``IsValidSignatureEncoding``.

There may be further simplifications that could be made by eliminating ``IsValidSignatureEncoding`` as well, but that would change when particular classes of invalid signatures are detected. That has the risk of causing unintended consensus changes, whereas the changes I've made in this PR can be verified locally to cause no change in behaviour.